src/soundio/sounddeviceportaudio.cpp

```diff
@@ -407,12 +413,28 @@ bool SoundDevicePortAudio::isOpen() const {
     return m_pStream.load() != nullptr;
 }
 
+void SoundDevicePortAudio::finishedCallback() {
+    // This callback is called by PortAudio when the process callback returns
+    // paAbort to indicate the stream has to finish. This is triggered in the
+    // SoundDevicePortAudio::close() method.
+    std::unique_lock lock(m_finishedMutex);
+    m_bFinished = true;
+    m_finishedCV.notify_one();
+}
+
+void SoundDevicePortAudio::waitForFinished() {
```

Comment:
This function should be inclined for a better readability.

Reply:
Sorry, what do you mean by that?

Reply:
Oh typo. I mean inlined. The function body should be closer to the other calls or the other calls can be moved here and the function renamed a bit. This will help to trace the control flow a bit better.